apps/web/pages/export.tsx

```diff
@@ -0,0 +1,253 @@
+import {
+  Button,
+  Card,
+  CircularProgress,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+} from '@usevenice/ui'
+
+import {
+  CheckCircleFilledIcon,
+  CodeIcon,
+  CopyTextIcon,
+  DownloadIcon,
+  SyncIcon,
+} from '@usevenice/ui/icons'
+
+import type {GetServerSideProps} from 'next'
+import Link from 'next/link'
+import {useEffect, useState} from 'react'
+import {PreviewResult, usePreviewQuery} from '../components/export'
+import {ExternalLink} from '../components/ExternalLink'
+import {PageHeader} from '../components/PageHeader'
+import {PageLayout} from '../components/PageLayout'
+
+const PREVIEW_LIMIT = 8
+const DOWNLOAD_LIMIT = 100
+
+interface ServerSideProps {
+  apiKey: string
+}
+
+export const getServerSideProps: GetServerSideProps<ServerSideProps> = async (
+  ctx,
+) => {
+  const {serverGetUser} = await import('../server')
+  const user = await serverGetUser(ctx)
+
+  const {z} = await import('@usevenice/util')
+  const apiKey = z.string().parse(user?.user_metadata?.['apiKey'])
+
+  return {
+    props: {apiKey},
+  }
+}
+
+export default function Page(props: ServerSideProps) {
+  const {apiKey} = props
+  const [selectedTable, selectTable] = useState<string>()
+  const preview = usePreviewQuery({table: selectedTable, limit: PREVIEW_LIMIT})
+  return (
+    <PageLayout title="Explore Data">
+      <PageHeader title={['Explore Data', 'CSV']} />
+      <div className="p-6">
+        <div className="grid grid-cols-[1fr_auto]">
+          <div className="flex items-center gap-4">
+            <Select value={selectedTable} onValueChange={selectTable}>
+              <SelectTrigger
+                className="max-w-[10rem]"
+                placeholder="Select a table…"
+              />
+              <SelectContent>
+                <SelectItem value="posting">Posting</SelectItem>
+                <SelectItem value="transaction">Transaction</SelectItem>
+                <SelectItem value="institution">Institution</SelectItem>
+                <SelectItem value="pipeline">Pipeline</SelectItem>
+                <SelectItem value="resource">Resource</SelectItem>
+              </SelectContent>
+            </Select>
+            {/* TODO fade in & out the loader */}
+            {preview.isFetching && <CircularProgress className="h-5 w-5" />}
+          </div>
+          <div className="flex gap-2">
+            <Button
+              variant="primary"
+              className="gap-1"
+              disabled={!selectedTable}
```

Comment:
Nit: may also want to disable it if `results.length` is `0`

Reply:
Good catch!

@iwasrobbed updated in 4f2a3d5 which makes sense for the button. However, for the inline link, it still looks like a link. I tried opacity treatment and it didn't feel right for an inline link. I tried changing the color to white when it's disabled (to make it like a text) which feels **really** wrong when it the data becomes available. I think I'll settle to what we have. wdyt?

Reply:
What you have works for me where it's just a disabled text link!